workbench-core/example/infrastructure/integration-tests/support/resources/dynamicAuthorization/group.ts

```diff
@@ -29,6 +31,11 @@ export default class Group extends Resource {
     return this._axiosInstance.get(`${this._api}/is-user-assigned/${userId}`);
   }
 
+  public async delete(): Promise<void> {
+    await super.delete();
+    this._clientSession.removeCleanupTask(`group-${this.id}`);
```

Comment:
It seems like this should be the default behavior for all delete requests. Can you put this logic in the `Resource` class?